Branch-SDK/Branch-SDK/BranchNotifications.h

```diff
@@ -0,0 +1,31 @@
+//
+//  BranchNotifications.h
+//  Branch-TestBed
+//
+//  Created by edward on 6/30/17.
+//  Copyright © 2017 Branch Metrics. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+@class Branch;
+
+@protocol BranchDelegate <NSObject>
```

Comment:
This is a nice addition to the SDK. I would at least add a delegate method like `branch:didOpenBranchUniversalObject:withLinkProperties:` and include a BUO and BranchLinkProperties in the notification payload. Using untyped data (params) without a data model is just bad app design in a typed language and should be discouraged, though I don't see any harm in exposing the raw params for those who want access to them for diagnostics or something. I would also argue for a lot more semantic distinction. This change adds new notification mechanisms as alternatives to a single callback block in the app delegate, which is a big improvement. But it's still confusing for lots of people (myself included) that the same callback used to notify users that a link was opened (which is generally all they care about) is called when no link is available. It's not a bad idea to give feedback that the SDK was initialized successfully, and no link is available. But calling the app when the SDK does a deferred deep link check on foreground and then notifying the app that nothing happened is not the best idea. And in any case, I think much more semantic distinction is required. I'd like to see distinct notifications and delegate methods for: initialization success; initialization failure; link open success; link open failure. The open events should only occur when a user actually opened a link without an error, and errors should be distinct events, not parameters in the success callback. A method like `didEncounterError:` is extremely common in Obj-C delegates. But maybe additional delegate methods and notifications would be a separate change.